ompi/mca/io/romio321/romio/adio/common/ad_read_coll.c

```diff
@@ -811,32 +810,33 @@ static void ADIOI_R_Exchange_data(ADIO_File fd, void *buf, ADIOI_Flatlist_node
 
     if (buftype_is_contig) {
 	j = 0;
-	for (i=0; i < nprocs; i++) 
+	for (i=0; i < nprocs; i++) { 
 	    if (recv_size[i]) {
 		MPI_Irecv(((char *) buf) + buf_idx[i], recv_size[i], 
 		  MPI_BYTE, i, myrank+i+100*iter, fd->comm, requests+j);
 		j++;
 		buf_idx[i] += recv_size[i];
 	    }
-    }
-    else {
+        }
+    } else {
 /* allocate memory for recv_buf and post receives */
 	recv_buf = (char **) ADIOI_Malloc(nprocs * sizeof(char*));
-	for (i=0; i < nprocs; i++) 
-	    if (recv_size[i]) recv_buf[i] = 
-                                  (char *) ADIOI_Malloc(recv_size[i]);
+	recv_buf[0] = (char *) ADIOI_Malloc(memLen);
```

Comment:
It looks like `memLen` will always be 0 when it gets to this point...?

Reply:
Oops, looks like I missed a line somewhere. Do you want me to add that, or should I revert this and just keep the `{}` fixes?

Reply:
Let's do just the `{}` fixes.